### ISSUE TITLE
WARNING: Mitre Technique ID 'T1550.002' not found in database

### ISSUE THREAD

**Issue by StefanSa** [wazuh/wazuh#9633]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.1.5-rev | ossec-analysisdt | Manager | Packages | openLeap-15.3 |

hi there,
Get quite a few of these error messages (T1550.002-Successful Remote Logon Detected - NTLM authentication, possible pass-the-hash attack):
```
2021/08/10 13:43:50 ossec-analysisd: WARNING: Mitre Technique ID 'T1550.002' not found in database.
2021/08/10 13:43:53 ossec-analysisd: WARNING: Mitre Technique ID 'T1550.002' not found in database.
2021/08/10 13:44:00 ossec-analysisd: WARNING: Mitre Technique ID 'T1550.002' not found in database.
2021/08/10 13:44:08 ossec-analysisd: WARNING: Mitre Technique ID 'T1550.002' not found in database.
2021/08/10 13:44:15 ossec-analysisd: WARNING: Mitre Technique ID 'T1550.002' not found in database.
2021/08/10 13:44:15 ossec-analysisd: WARNING: Mitre Technique ID 'T1550.002' not found in database.
```
I wanted to recreate the mitre.db `python3 mitredb.py`, but i also get an error message:
```
Traceback (most recent call last):
  File "mitredb.py", line 777, in <module>
    main(args.database)
  File "mitredb.py", line 769, in main
    gid = grp.getgrnam("wazuh").gr_gid
KeyError: "getgrnam(): name not found: 'wazuh'"
```

**Comment by StefanSa** [comment]
with the latest version of `enterprise-attack.json`(april update),  i get this error message when creating the` mitre.db`:
```
cd ../tools/mitre && /var/ossec/framework/python/bin/python3 mitredb.py -d /var/ossec/var/db/mitre.db
'kill_chain_phases'
Deleting /var/ossec/var/db/mitre.db
```

**Comment by NicoGuini** [comment]
TL;DR;
This works for upcoming version of Wazuh 4.2


Wazuh version 4.1 has no capabilities to parse the latests version of Mitre Matrix JSON. This latest JSON has techniques, subtechniques and a buch of changes that Wazuh 4.1 can't parse properly. Evend the generated DB has a completly change of structure, and how it's consumed also was changed.

Next upcoming version of Wazuh, 4.2 will have the capability to parse it properly. Internal tests made to confirm this works as expected.

**Comment by StefanSa** [comment]
hola @NicoGuini 
any ETA for v4.2 ?

**Comment by NicoGuini** [comment]
Hi @StefanSa ! Wazuh v4.2 just got released!

**Comment by NicoGuini** [comment]
Hi @StefanSa .

## Pre-requisites:
Update Manager and Workers to V4.2


## Steps
The steps to update the mitre.db should be the following:
```
# wazuh_home=/var/ossec
# python3 tools/mitre/mitredb.py -d ./mitre.db  (Compile mitre.db)
# rsync ./mitre.db $wazuh_home/var/db/mitre.db (Copy mitre.db)
# service wazuh-manager restart (Restart manager, or cluster)
```

**Comment by StefanSa** [comment]
Hi @NicoGuini
Thanks for your help, that wasn't the trick.
A Mtre ID is now also being claimed.
```
2021/08/30 15:38:38 wazuh-analysisd: WARNING: Mitre Technique ID 'T1550.002' not found in database.
2021/08/30 15:38:38 wazuh-analysisd: WARNING: Mitre Technique ID 'T1078' not found in database.
```
Question, how do you get the `last git rules set` and the `August mitre.db` update to work without an error message?
Thanks for any help
Stefan

**Comment by NicoGuini** [comment]
@StefanSa you can download ZIP from here: https://attack.mitre.org/resources/versions/

Direct link: https://github.com/mitre/cti/releases/tag/ATT%26CK-v9.0

**Comment by StefanSa** [comment]
Hi @NicoGuini & @jcruzlp 
The whole thing is a bit confused.
Also with this file, April update, the Mitre ID's `T1078`and `T1550.002` are not detected with wazuh v4.2.
The same is true in the Wazuh app v4.2, here in the framework all Mitre ID's with sub-items are not showing.
![Mitre02](https://user-images.githubusercontent.com/6105075/131492565-c414f5e7-12b0-4ec1-8c0e-65ac8fdcd3f9.PNG)

The whole thing looks completely different in wazuh-server and wazuh-app in version v4.3.
Here there are no error messages about missing Mitre ID's in the database, also in the framework all Mitre ID with subitems are shown correctly.
![Mitre01](https://user-images.githubusercontent.com/6105075/131492985-5dc56f72-ec8f-460c-a288-1ebb70a31e5c.PNG)

For me it currently looks like that wazuh in version v4.2 can not work correctly with the Mitre update from April.
Can any of the dev's confirm this before we continue to search endlessly for the bug here ?

Thanks for your help and time.
Stefan

**Comment by adampielak** [comment]
Hi, 
i have same problem Wazuh v4.2.4:
```
grep Mitre /var/ossec/logs/ossec.log |awk '{print $8}'|sort|uniq
'T1021.001'
'T1021.004'
'T1070.004'
'T1078.002'
'T1110.001'
'T1499.001'
'T1543.003'
'T1548.003'
'T1550.002'
'T1562.001'
'T1565.001'
'T1570'
'T1595.002'
```

**Comment by iasdeoupxe** [comment]
Newer similar issue: #18325
